### PR TITLE
Skip clearing guarded DIMMs

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**
The commit modifies script to skip clearing operation status for the DIMMs only if the DIMM is guarded.

The change has been done as operational status was being cleared for all the DIMMs at start up which should not be a case in case the DIMM has been marked as guarded.

### CODE REVIEW

File: scripts/clear-all-fault-leds.sh [hunk 1]
```diff
@@ -68,11 +86,18 @@ then
...
+	echo "$line" | grep "dimm" >/dev/null
+	rc=$?
+	if [ $rc -eq 0 ]; then
+	    if [[ ${guardedDimms[@]} =~ $line ]]; then
+		continue
+	    fi
+	fi
         busctl set-property xyz.openbmc_project.Inventory.Manager "$line" xyz.openbmc_project.State.Decorator.OperationalStatus Functional b "$action";
     done
 else
```
Comment (jinuthomas) [inline scripts/clear-all-fault-leds.sh:103]: The changes need to be there in else part for excluded groups also

**Review by jinuthomas — CHANGES_REQUESTED** [review]

File: scripts/clear-all-fault-leds.sh (hunk 1 above)
Comment (SunnySrivastava1984) [reply scripts/clear-all-fault-leds.sh:103]: Done

File: scripts/clear-all-fault-leds.sh [hunk 2]
```diff
@@ -68,23 +86,37 @@ then
         GetSubTreePaths sias "/xyz/openbmc_project/inventory" 0 1 "xyz.openbmc_project.State.Decorator.OperationalStatus" \
         | sed  's/ /\n/g' | tail -n+3 | awk -F "\"" '{print $2}' | while read -r line
     do
-        echo "$line" | grep "powersupply\|dimm\|dcm\|fan" >/dev/null
+        echo "$line" | grep "powersupply\|dcm\|fan" >/dev/null
```
Comment (spinler) [inline scripts/clear-all-fault-leds.sh:89]: so this still won't set functional back to true for PSs and fans?  I forget, why is that?

Comment (jinuthomas) [reply scripts/clear-all-fault-leds.sh:89]: we will ultimately try and fix the PS and fans. this fix is for DIMMs only

**Comment by geissonator** [comment]
add to approvelist

**Comment by SunnySrivastava1984** [comment]
No longer needed.

**Pull request closed without merging**